components/datetime/src/error.rs

```diff
@@ -28,6 +29,12 @@ pub enum DateTimeFormatterLoadError {
     /// such as skeleton patterns or calendar conversions.
     #[displaydoc("{0}")]
     Data(DataError),
+    /// The calendar preference is not supported
+    #[displaydoc("Unsupported calendar {0:?}")]
+    UnsupportedCalendar(CalendarAlgorithm),
+    /// The AnyCalendar is not supported
+    #[displaydoc("Unsupported calendar {0:?}")]
+    UnsupportedAnyCalendar(AnyCalendarKind),
```

Comment:
:frowning_face: but we can fix this up after this lands. Make sure we have an issue.